Teleportation/Teleportation.ipynb

```diff
@@ -223,34 +222,79 @@
     "  <summary><b>Need a hint? Click here</b</summary>\n",
     "  You may find your answers for 1.5 and 1.6 useful\n",
     "</details>"
-   ]
+   ],
+   "metadata": {}
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
-   "outputs": [],
    "source": [
     "open Microsoft.Quantum.Diagnostics;\n",
     "\n",
     "operation Run_StandardTeleport () : String {\n",
     "    // ...\n",
     "    return \"Success!\";\n",
     "}"
-   ]
+   ],
+   "outputs": [],
+   "metadata": {}
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
-   "outputs": [],
    "source": [
     "%simulate Run_StandardTeleport"
-   ]
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Task 1.8. Entanglement swapping\n",
+    "\n",
+    "Alice and Bob, independently from each other, each hold an entangled qubit pair in the state $|\\Phi^{+}\\rangle = \\frac{1}{\\sqrt{2}} \\big(|00\\rangle + |11\\rangle\\big)$. They hand off one part of their pair to Charlie.\n",
+    "\n",
+    "Charlie can now teleport the state of Alice's qubit onto Bob's qubit, thus teleporting the entanglement. \n",
+    "Just like in \"standard\" teleportation, Bob still needs to apply the reconstruction steps - based on Charlie's measurement results - to the other qubit in his possession.\n",
+    "\n",
+    "The practical outcome is that the entangled state $|\\Phi^{+}\\rangle = \\frac{1}{\\sqrt{2}} \\big(|00\\rangle + |11\\rangle\\big)$. now spans across Alice's and Bob's qubits - the ones that they didn't send to Charlie. They are now maximally entangled even though they never interacted in the first place!\n",
+    "\n",
+    "**Goal:** \n",
+    "\n",
+    "Using state teleportation, entangle the **other part** of Alice's entangled pair (the one you do not have access to!) with one part (`qBob2`) of Bob's input qubit pair.\n",
+    "\n",
+    "**Outputs:**\n",
+    "\n",
+    "A tuple of two operations.\n",
+    "\n",
+    "The first operation will take two qubits as input (one belonging to Alice and one to Bob), and produce a joint measurement result as a two-bit integer.\n",
+    "\n",
+    "The second operation will take Bob's other qubit and Charlie's 2-bit integer as input, and should use the integer to adjust the state of the qubit to reconstruct the state $|\\Phi^{+}\\rangle$ across the now entangled Alice's and Bob's qubits.\n",
+    "\n",
+    "<details>\n",
+    "  <summary><b>Need a hint? Click here</b</summary>\n",
+    "  You may find your answers for 1.2 and 1.3 useful, as similar steps are needed here\n",
+    "</details>"
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "%kata T18_EntanglementSwapping\n",
+    "\n",
+    "operation EntanglementSwapping () : ((Qubit, Qubit) => Int, (Qubit, Int) => Unit) {\n",
+    "    // ...\n",
+    "    fail \"Not implemented yet\";\n",
```

Comment:
We don't need a fail in the notebook version of the task, since the notebook cells don't have to compile successfully for the previous ones to work, unlike the Q# projects